spec/sec-regexp-regular-expression-objects-patch.html

```diff
@@ -104,50 +104,107 @@ <h1>Runtime Semantics: BackreferenceMatcher ( _n_, _direction_ )</h1>
   </emu-clause>
 </emu-clause>
 
+<emu-clause id="sec-regexp-abstract-operations">
+  <h1>RegExp Abstract Operations</h1>
+
+  <emu-clause id="sec-matchoptions-records">
+    <h1><ins>MatchOptions Records</ins></h1>
+    <ins>
+    <p>A MatchOptions is a Record value used to encapsulate options for a regular expression matching operation.</p>
+    <p>MatchOptions Records have the fields listed in <emu-xref href="#table-matchoptions-record"></emu-xref>.</p>
+    </ins>
+    <emu-table id="table-matchoptions-record" caption="MatchOptions Record Fields">
+      <table>
+        <tbody>
+          <tr>
+            <th>Field Name</th>
+            <th>Value</th>
+            <th>Meaning</th>
+          </tr>
+          <tr>
+            <td>[[Captures]]</td>
+            <td>`"strings"` or `"indices"`</td>
+            <td>Whether the operation should capture substrings or the start/end indices of the capture.</td>
+          </tr>
+        </tbody>
+      </table>
+    </emu-table>
+  </emu-clause>
+
+  <emu-clause id="sec-frommatchoptions" aoid="FromMatchOptions">
+    <h1><ins>FromMatchOptions ( _Options_ )</ins></h1>
+    <p>
+      When the abstract operation FromMatchOptions is called with MatchOptions _Options_, the following steps are taken:
+    </p>
+    <emu-alg>
+    1. If _Options_ is *undefined*, return *undefined*.
+    1. Let _obj_ be ObjectCreate(%ObjectPrototype%).
```

Comment:
any reason not to make this ObjectCreate(null)?

Reply:
This is being passed to a user-overridable function. They might want to be able to use `.hasOwnProperty` et al.

Reply:
Not sure if we should encourage the bad practice of directly calling Object.prototype methods on objects :-)

Reply:
We only use `ObjectCreate(*null*)` in 4 places in the spec (unscopeables, realm intrinsics, mapped arguments objects in non-strict functions with simple parameter lists, and the `groups` object on a regex match result). We call `ObjectCreate(%ObjectPrototype%)` in 13 places, including `FromPropertyDescriptor`, which this function emulates.

Reply:
I think it would still be a good question to bring up in committee, since this will establish the precedent for all future options objects in the language, but i'll resolve this for now as-is.